src/base/ApplicationDynamoDBTable.ts

```diff
@@ -100,29 +105,110 @@ export class ApplicationDynamoDBTable extends Resource {
-        roleArn: ApplicationDynamoDBTable.createAutoScalingRole(
+      capacityType,
+      prefix,
+      dynamoDB.arn,
+      tags
+    );
+
+    // create an auto scaling policy for the table
+    ApplicationDynamoDBTable.createAutoScalingPolicy(
+      scope,
+      roleArn,
+      'table',
+      capacityType,
+      config.min,
+      config.max,
+      config.tracking,
+      dynamoDB.name
+    );
+
+    // create an auto scaling policy for each global secondary index
+    if (globalSecondaryIndexes.length) {
+      globalSecondaryIndexes.forEach((gsIndex) => {
+        // min capacity is defined by the global secondary index
+        // max capacity is inherited from the table auto scaling config
+        // TODO: if we want this to be configurabe per index, we'll need to extend the third-party interface
+        const minCapacity =
+          capacityType === ApplicationDynamoDBTableCapacityType.Read
+            ? gsIndex.readCapacity
+            : gsIndex.writeCapacity;
+
+        // create an auto scaling policy for each index
+        ApplicationDynamoDBTable.createAutoScalingPolicy(
           scope,
+          roleArn,
+          'index',
           capacityType,
-          prefix,
-          dynamoDB.arn,
-          tags
-        ),
+          minCapacity,
+          config.max,
+          config.tracking,
+          dynamoDB.name,
+          gsIndex.name
+        );
+      });
+    }
+  }
+
+  /**
+   * sets up autoscaling policy for a table or an index
+   * @param scope
+   * @param roleArn
+   * @param policyTarget
+   * @param capacityType
+   * @param minCapacity
+   * @param maxCapacity
+   * @param tracking
+   * @param tableName
+   * @param indexName
+   * @private
+   */
+  private static createAutoScalingPolicy(
+    scope: Construct,
+    roleArn: string,
+    policyTarget: 'table' | 'index',
+    capacityType: ApplicationDynamoDBTableCapacityType,
+    minCapacity: number,
+    maxCapacity: number,
+    tracking: number,
+    tableName: string,
+    indexName?: string
+  ): void {
+    let resourceId = `table/${tableName}`;
+
+    // if we're targeting an index, the resource id must reflect that
+    if (policyTarget === 'index') {
+      if (indexName) {
+        resourceId += `/index/${indexName}`;
+      } else {
+        throw new Error(
+          'you must specify an indexName when creating an index auto scaling policy'
+        );
+      }
+    }
+
+    const targetTracking = new AppautoscalingTarget(
+      scope,
+      `${capacityType}_${policyTarget}_target`,
+      {
+        maxCapacity,
+        minCapacity,
+        resourceId,
+        scalableDimension: `dynamodb:${policyTarget}:${capacityType}Units`,
+        roleArn: roleArn,
         serviceNamespace: 'dynamodb',
-        dependsOn: [dynamoDB],
+        // TODO: the interface doesn't seem to support dependsOn any longer
+        // can we drop this?
+        //dependsOn: [dynamoDB],
```

Comment:
Depends on like lifecycle above are baseline terraform properties. Perhaps the cdk renamed them?